add min changes to get GNU_HASH recognized so anal bfd's don't reject
objects that contain this section

building a lib.so with binutils-2.18 and GNU_HASH and then link with
binutils-2.17 will result in failure on some (like x86_64) systems

http://sourceware.org/ml/binutils/2007-08/msg00198.html

--- include/elf/common.h
+++ include/elf/common.h
@@ -338,6 +338,7 @@
 #define SHT_LOOS	0x60000000	/* First of OS specific semantics */
 #define SHT_HIOS	0x6fffffff	/* Last of OS specific semantics */
 
+#define SHT_GNU_HASH	0x6ffffff6	/* GNU style symbol hash table */
 #define SHT_GNU_LIBLIST	0x6ffffff7	/* List of prelink dependencies */
 
 /* The next three section types are defined by Solaris, and are named
--- bfd/elf.c
+++ bfd/elf.c
@@ -1822,6 +1822,7 @@ bfd_section_from_shdr (bfd *abfd, unsign
     case SHT_FINI_ARRAY:	/* .fini_array section.  */
     case SHT_PREINIT_ARRAY:	/* .preinit_array section.  */
     case SHT_GNU_LIBLIST:	/* .gnu.liblist section.  */
+    case SHT_GNU_HASH:		/* .gnu.hash section.  */
       return _bfd_elf_make_section_from_shdr (abfd, hdr, name, shindex);
 
     case SHT_DYNAMIC:	/* Dynamic linking information.  */
